src/Workspaces/Core/Portable/Workspace/Solution/ProjectId.cs

```diff
@@ -28,8 +28,12 @@ private ProjectId(string debugName)
             _debugName = debugName;
         }
 
-        internal ProjectId(Guid guid, string debugName)
+        public ProjectId(Guid guid, string debugName = null)
```

Comment:
I'm going to use the name CreateForSerialization so that I can unblock this PR.  But I think this is a decision we should revisit.  APIs should be named for what they do, not one of the scenarios they serve.